demos/object_detection_demo/main.cpp

```diff
@@ -0,0 +1,257 @@
+        std::vector<std::string> labels;
+        if (!FLAGS_labels.empty())
+            labels = DetectionModel::loadLabels(FLAGS_labels);
+
+        std::unique_ptr<ModelBase> model;
+        if (FLAGS_mt=="ssd")
+        {
+            model.reset(new ModelSSD(FLAGS_m, (float)FLAGS_t, FLAGS_auto_resize, labels));
+        }
+        else if (FLAGS_mt=="yolo")
+        {
+            model.reset(new ModelYolo3(FLAGS_m,(float)FLAGS_t, FLAGS_auto_resize, (float)FLAGS_iou_t, labels));
+        }
+        else
+        {
+            slog::err << "No model type or invalid model type (-mt) provided: " + FLAGS_mt << slog::endl;
+            return -1;
+        }
+
+        InferenceEngine::Core core;
+        PipelineBase pipeline(std::move(model),
+            ConfigFactory::getUserConfig(FLAGS_d, FLAGS_l, FLAGS_c, FLAGS_pc, FLAGS_nireq, FLAGS_nstreams, FLAGS_nthreads),
+            core);
+        Presenter presenter;
+
+        auto startTimePoint = std::chrono::steady_clock::now();
+        while (true){
+            int64_t frameNum;
+            if (pipeline.isReadyToProcess()) {
+                //--- Capturing frame. If previous frame hasn't been inferred yet, reuse it instead of capturing new one
+                curr_frame = cap->read();
+                if (curr_frame.empty()) {
+                    if (!frameNum) {
+                        throw std::logic_error("Can't read an image from the input");
+                    }
+                    else {
+                        // Input stream is over
+                        break;
+                    }
+                }
+
+                frameNum = pipeline.submitImage(curr_frame);
+            }
+
+            //--- Waiting for free input slot or output data available. Function will return immediately if any of them are available.
+            pipeline.waitForData();
+
+            int key = 0;
+
+            //--- Checking for results and rendering data if it's ready
+            //--- If you need just plain data without rendering - cast result's underlying pointer to DetectionResult*
+            //    and use your own processing instead of calling renderDetectionData().
+            std::unique_ptr<ResultBase> result;
+            while (result = pipeline.getResult()) {
+                cv::Mat outFrame = DefaultRenderers::renderDetectionData(result->asRef<DetectionResult>());
+                //--- Showing results and device information
+                if (!outFrame.empty()) {
+                    presenter.drawGraphs(outFrame);
+                    pipeline.getMetrics().paintMetrics(outFrame, { 10,22 }, 0.6);
+                    if (!FLAGS_no_show)
+                    {
+                        cv::imshow("Detection Results", outFrame);
+
+                        // Showing frame in window and storing key pressed if key variable doesn't contain key code yet
+                        if (key) {
+                            cv::waitKey(1);
```

Comment:
well, I was trying to avoid introducing extra flags variables to cancel main cycle. But now I see that this solution is not good either, I've changed it